### ISSUE TITLE
Build Simple Interface for Staff to set up a session

### ISSUE THREAD

**Issue by stmball** [rtreharne/fizzynewt#3]
Needs:

- Backend models for Student, Log, Module (Maybe a Lecture model to aggregate logs easily?)
- Add API routes for creation of Modules, getting a list of modules (for the form)
- Test using fake students and modules
- Generation and logging of random codes (adjective + animal). @rtreharne def best to do this cause it's a copy and paste job from the last app!
- Frontend page for form to select Module, submit to API. Validation???
- Tests tests tests. Let's do this properly. Maybe CI + CD if we can figure that out.

**Comment by rtreharne** [comment]
Brill. Thanks for writing up these tasks as issues @stmball. Will start on the models and update you with progress.

**Comment by rtreharne** [comment]
Just created the following models:
    - Create model Institution
    - Create model SessionType
    - Create model UserType
    - Create model User
    - Create model Log
    - Create model IdFormat
    - Create model EmailSubstring
    - Create model CourseCodeFormat

**Comment by stmball** [comment]
Not sure I can see any of those changes - which branch did you push them to?

I've created a new branch `simple_staff` for all the frontend stuff for the simple code generation. It doesn't have any of the backend or frontend talking to backend logic in yet although I may get started on this after going for a run!

**Comment by rtreharne** [comment]
I pushed them to "main". Can you see it? Will merge my changes to "simple_staff" and re-push.

**Comment by stmball** [comment]
Yep - I see now! It's probably best to use the rule of "new function, new branch" - helps keep things more organised + stops us pushing test branches to the main project. Want to talk today?

**Comment by rtreharne** [comment]
OK. I've created a View and Serializer for the Log model (although not sure I should be calling it log as the model represents a unique code object - a "fizzy newt" code). Am I right in thinking that you can make a post to this too?
